src/pwm.rs

```diff
@@ -258,10 +370,59 @@ macro_rules! hal {
                         .set_bit()
                 );
 
-                unsafe { mem::MaybeUninit::uninit().assume_init() }
+                Pwm {
+                    clk: clk,
+                    _tim: PhantomData,
+                    _channels: PhantomData
+                }
+            }
+            
+            pwm_impl!(
+                $TIMX, $timX, (0, 1, 2, 3), (C1, C2, C3, C4);
+                $TIMX, $timX, (0, 1, 2), (C1, C2, C3);
+                $TIMX, $timX, (0, 1, 2), (C1, C2, C4);
+                $TIMX, $timX, (0, 1, 2), (C2, C3, C4);
+                $TIMX, $timX, (0, 1), (C1, C2);
+                $TIMX, $timX, (0, 1), (C1, C3);
+                $TIMX, $timX, (0, 1), (C1, C4);
+                $TIMX, $timX, (0, 1), (C2, C3);
```

Comment:
That's quite disturbing. I'd prefer an `pub enum Channel { C1, C2, C3, C4 }` as `hal::Pwm::Channel` and then just having `Channel::C2` to manage `C2`

Reply:
> `Pwm::into_channels(self) -> PWMCHANNELS`

or `split()`

Reply:
@TeXitoi I think this comment is OBE based on the updated code I pulled from @burrbull. Please reopen if not.